Replace fvdl with ffx, allow test without install

djkoloski opened the pull request:
Along with replacing fvdl uses with the equivalent ffx commands, this also switches from using the install path for libstd-*.so and libtest-*.so to using the build directory (now passed on the command line). The user no longer needs to run x.py install before running tests now, and the correct libstd and libtest are detected on run instead of startup so the test runner can handle recompilations after starting the testing environment.

r? @tmandry

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5293920925/jobs/9582654860) [(plain)](https://github.com/rust-lang/rust/commit/6e1f33df98f23fed6426d26c3d6ce367bb9637f3/checks/14333157881/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v3' (SHA:c85c95e3d7251135ab7dc9ce3241c5835cc595a9)
Download action repository 'rust-lang/simpleinfra@master' (SHA:dde6cb90e68f892d1d4b98f67afc9f2bd1164dcd)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  CI_JOB_NAME: mingw-check-tidy
---
Building wheels for collected packages: reuse
  Building wheel for reuse (pyproject.toml): started
  Building wheel for reuse (pyproject.toml): finished with status 'done'
  Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=180116 sha256=351235b2326fb4db7a18e257e13ce7896c5f77339521e2c2612e71e154800a19
  Stored in directory: /tmp/pip-ephem-wheel-cache-0jeemq55/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
  Attempting uninstall: setuptools
    Found existing installation: setuptools 59.6.0
    Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
Successfully tagged rust-ci:latest
Built container sha256:24761659b62f7f7f7b42a753e07be2ba7a5a1f4bb0a39912742769dc13027d8d
Uploading finished image to https://ci-caches.rust-lang.org/docker/e51156f19850ce886cec818e46dc2a021e0aa7c270d15673e8fe74cd8522fc8ac3995109aebb688ee49ed586735ed4cf5f8c06d44c48298fa09c35ae4b082281

<botocore.awsrequest.AWSRequest object at 0x7f26032fe110>
gzip: stdout: Broken pipe
xargs: docker: terminated by signal 13
[CI_JOB_NAME=mingw-check-tidy]
[CI_JOB_NAME=mingw-check-tidy]
---
##[endgroup]
fmt check
tidy check
tidy: Skipping binary file check, read-only filesystem
##[error]tidy error: /checkout/src/ci/docker/scripts/fuchsia-test-runner.py:261: trailing whitespace
##[error]tidy error: /checkout/src/ci/docker/scripts/fuchsia-test-runner.py:284: TODO is used for tasks that should be done before merging a PR; If you want to leave a message in the codebase use FIXME
##[error]tidy error: /checkout/src/ci/docker/scripts/fuchsia-test-runner.py:297: TODO is used for tasks that should be done before merging a PR; If you want to leave a message in the codebase use FIXME
some tidy checks failed
```

</details>

tmandry commented on src/ci/docker/scripts/fuchsia-test-runner.py:
```diff
@@ -502,6 +451,19 @@ def run(self, args):
+
+        libstd_name = os.path.basename(libstd_paths[0])
+        libtest_name = os.path.basename(libtest_paths[0])
+
```
What about pretending the libs path here so the template doesn't have to also know about it?

tmandry commented on src/ci/docker/scripts/fuchsia-test-runner.py:
```diff
@@ -275,87 +256,63 @@ def start(self):
-        # Start emulator (this will generate the vdl output)
-        self.log_info("Starting emulator...")
+        # FIXME: condition --accel hyper on target arch matching host arch
         subprocess.check_call(
```
Should this be removed?

djkoloski replied:
I think we want to leave this in, `--accel hyper` is almost always what we want. It just might not always work. I haven't tested what happens if we `--accel hyper` when the architectures don't match.

tmandry commented on src/ci/docker/scripts/fuchsia-test-runner.py:
```diff
@@ -275,87 +256,63 @@ def start(self):
+
+        # Start emulator
+        self.log_info("Starting emulator...")
+        product_bundle = "terminal.qemu-" + self.triple_to_arch(self.target)
```
remove?

djkoloski replied:
Not sure what you mean by this.

tmandry replied:
nevermind, this was supposed to be on an earlier diff.

djkoloski committed:
Replace fvdl with ffx, allow test without install

Along with replacing fvdl uses with the equivalent ffx commands, this
also switches from using the install path for libstd-*.so and
libtest-*.so to using the build directory (now passed on the command
line). The user no longer needs to run x.py install before running tests
now, and the correct libstd and libtest are detected on run instead of
startup so the test runner can handle recompilations after starting the
testing environment.

tmandry commented:
looks like there's an unintentional submodule change

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5293997705/jobs/9582819805) [(plain)](https://github.com/rust-lang/rust/commit/c7d9df8cc37789c4f30e7cb4b8b485e64bca02cc/checks/14333375032/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v3' (SHA:c85c95e3d7251135ab7dc9ce3241c5835cc595a9)
Download action repository 'rust-lang/simpleinfra@master' (SHA:dde6cb90e68f892d1d4b98f67afc9f2bd1164dcd)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  CI_JOB_NAME: mingw-check-tidy
---
Building wheels for collected packages: reuse
  Building wheel for reuse (pyproject.toml): started
  Building wheel for reuse (pyproject.toml): finished with status 'done'
  Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=180116 sha256=351235b2326fb4db7a18e257e13ce7896c5f77339521e2c2612e71e154800a19
  Stored in directory: /tmp/pip-ephem-wheel-cache-r4yg8ozn/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
  Attempting uninstall: setuptools
    Found existing installation: setuptools 59.6.0
    Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
Successfully tagged rust-ci:latest
Built container sha256:821d469d105677150c5ea6fc0574bda2947f233f440dbf79da73e11b7cc303f2
Uploading finished image to https://ci-caches.rust-lang.org/docker/e51156f19850ce886cec818e46dc2a021e0aa7c270d15673e8fe74cd8522fc8ac3995109aebb688ee49ed586735ed4cf5f8c06d44c48298fa09c35ae4b082281

<botocore.awsrequest.AWSRequest object at 0x7f94f7796110>
gzip: stdout: Broken pipe
xargs: docker: terminated by signal 13
[CI_JOB_NAME=mingw-check-tidy]
[CI_JOB_NAME=mingw-check-tidy]
---
##[endgroup]
fmt check
tidy check
tidy: Skipping binary file check, read-only filesystem
##[error]tidy error: /checkout/src/ci/docker/scripts/fuchsia-test-runner.py:261: trailing whitespace
some tidy checks failed
```

</details>

tmandry commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 4c6fd7594d6526c7ad58d3bc16e42c0dc538017a has been approved by `tmandry`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=tmandry 4c6fd7594d6526c7ad58d3bc16e42c0dc538017a -->
<!-- homu: {"type":"Approved","sha":"4c6fd7594d6526c7ad58d3bc16e42c0dc538017a","approver":"tmandry","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

tmandry committed:
Fix --rust-build flag in docs

tmandry committed:
Fix edit fail

tmandry commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 12c6f1d9588cd8775e13c6e85fa28794613f7446 has been approved by `tmandry`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=tmandry 12c6f1d9588cd8775e13c6e85fa28794613f7446 -->
<!-- homu: {"type":"Approved","sha":"12c6f1d9588cd8775e13c6e85fa28794613f7446","approver":"tmandry","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
